src/main/java/org/broadinstitute/hellbender/tools/sv/cluster/CanonicalSVLinkage.java

```diff
@@ -222,12 +274,31 @@ private static int getLength(final SVCallRecord record, final int missingInserti
             return record.getLength() == null ? missingInsertionLength : Math.max(record.getLength(), 1);
         } else if (record.getType() == GATKSVVCFConstants.StructuralVariantAnnotationType.BND) {
             return record.getPositionB() - record.getPositionA() + 1;
+        } else if (record.getType() == GATKSVVCFConstants.StructuralVariantAnnotationType.CPX) {
+            return getLengthComplex(record, missingInsertionLength);
         } else {
             // TODO lengths less than 1 shouldn't be valid
             return Math.max(record.getLength() == null ? 1 : record.getLength(), 1);
         }
     }
 
+    /**
+     * Returns length to be used for complex variant matching
+     */
+    private static int getLengthComplex(final SVCallRecord record, final int missingInsertionLength) {
+        if (!record.isIntrachromosomal() || record.getPositionA() == record.getPositionB()) {
+            // Insertion types use the sum of cpx intervals, or the "missing" length if there are none
+            if (record.getComplexEventIntervals().isEmpty()) {
+                return missingInsertionLength;
```

Comment:
Codecov is alerting me that you need a unit test for this case - although, shouldn't this throw an error if CPX_INTERVALS is empty for a CPX SV?

Reply:
Removed this method. I didn't want to make any assumptions about whether CPX_INTERVALS is present, but I do think we need to draw up a spec for them.